minetest/minetest
# Calculate sky color using raw brightness without gamma correction

Fixes #10258

In the calculation of sky color I've replace light-curve-corrected perceptive brightness in night/dawn/day detection rules with light-curve-agnostic linear brightness based on `time_of_day`.

Time points for dawn are preserved at 4:45 to 5:00, symmetric time for dusk.

## To do

This PR is Ready for Review.

## How to test

1. Add snippet from #10358 to devtest
2. Change light curve parameters (lighting_alpha, lighting_beta, lighting_boost, display_gamma ...) to any valid value (e.g. bright nights etc.)
3. Set time_speed to 576
4. Enter the game and set time to 4:00
5. Observe the transition of sky color from blue to green to red from 4:45 to 5:00, regardless of the light parameters.
